fcrepo-client-impl/src/main/java/org/fcrepo/client/impl/FedoraRepositoryImpl.java

```diff
@@ -143,7 +153,7 @@ public FedoraDatastream createDatastream(final String path, final FedoraContent
             final int statusCode = status.getStatusCode();
 
             if (statusCode == SC_CREATED) {
-                return getDatastream(path);
+                return getDatastream(prependTransactionId(path));
```

Comment:
Although this scenario is recognized and ignored in the implementation of `prependTransactionId(..)`, is there ever a case where this apparently redundant call to `prependTransactionId(..)` is not redundant? If not, I would suggest removing the update to this line.